src/Environment.php

```
@@ -477,15 +528,29 @@ public static function context(): ?ContextInterface {
-      // Load known contexts from the constant (no filesystem scanning).
-      foreach (self::CONTEXTS as $class) {
-        static::addContext(new $class());
+      static::$contexts = [];
+
+      $instances = array_merge(self::CONTEXTS, $additional);
+
+      foreach ($instances as $instance) {
+        $instance = is_string($instance) ? new $instance() : $instance;
+
+        if (!($instance instanceof ContextInterface)) {
+          throw new \InvalidArgumentException('The provider must implement ContextInterface');
```

PHPMD (2.15.0)

Missing class import via use statement (line '547', column '21'). (undefined)

(MissingImport)